app/frontend/wallet/shelley/transaction/splitTxOutput.ts

```diff
@@ -0,0 +1,16 @@
+import {TxOutput} from '../../types'
+import {computeMinUTxOLovelaceAmount} from './utils'
+
+export const splitTxOutput = (output: TxOutput, maxOutputTokens: number): TxOutput[] => {
+  const nOutputs = Math.ceil(output.tokenBundle.length / maxOutputTokens)
+  const outputs: TxOutput[] = []
+  for (let i = 0; i < nOutputs; i++) {
+    const tokenBundle = output.tokenBundle.slice(i * maxOutputTokens, (i + 1) * maxOutputTokens)
+    outputs.push({
+      ...output,
+      coins: computeMinUTxOLovelaceAmount(tokenBundle),
```

Comment:
I guess this is really doable only for a change output. You wouldn't really want to overwrite the amount being sent out for a non-change output. I would at least rename the function to splitTxChangeOutput to make it clear it applies to change outputs only and perhaps leave here a clarifying comment warning that this doesn't preserve original amount being sent at all